sdk/core/azure-core-test/TestProxyMigrationGuide.md

```diff
@@ -0,0 +1,237 @@
+could instead use the `BODY_REGEX` sanitizer type:
+
+```java
+customSanitizer.add(new TestProxySanitizer("(?<=\\/\\/)[a-z]+(?=(?:|-secondary)\\.table\\.core\\.windows\\.net)", "REDACTED", TestProxySanitizerType.URI));
+```
+In the snippet above, any storage endpoint URIs that match the specified URL regex will have their account name replaced with
+"REDACTED". A request made to `https://tableaccount-secondary.table.core.windows.net` will be recorded as being
+made to `https://REDACTED-secondary.table.core.windows.net`, and URLs will also be sanitized in bodies and headers.
+
+For more details about sanitizers and their options, please refer to [TestProxySanitizer][test_proxy_sanitizer].
+
+#### Note regarding body matching
+
+In the old testing system, request and response bodies weren't compared in playback mode by default in
+most packages. The test proxy system enables body matching by default, which can introduce failures for tests that
+passed in the old system. For example, if a test sends a request that includes the current Unix time in its body, the
+body will contain a new value when run in playback mode at a later time. This request might still match the recording if
+body matching is disabled, but not if it's enabled.
+
+Body matching can be turned off with the test proxy by calling the `setComparingBodies` method from
+[TestProxyRequestMatcher][test_proxy_matcher] on the Interceptor Manager for that particular test.
+
+## Migrate management-plane tests
+
+For management-plane packages, test classes should inherit from [TestProxyTestBase][test_proxy_base].
+
+The rest of the information in this guide applies to management-plane packages as well, except for possible specifics
+regarding test resource deployment.
+
+## Next steps
+
+Once your tests have been migrated to the test proxy, they can also have their recordings moved out of the
+`azure-sdk-for-java` repo. Refer to the [recording migration guide][recording_migration] for more details.
+
+After recordings are moved, you can refer to the instructions in [`TestProxyMigration.md`][test_proxy_migration] to manage them.
+
+## Advanced details
+
+### What does the test proxy do?
+
+The basic idea is to have a test server that sits between the client being tested and the live endpoint. 
+Instead of mocking out the communication with the server, the communication can be redirected to a test server.
+
+For example, if an operation would typically make a GET request to
+`https://fakeazsdktestaccount.table.core.windows.net/Tables`, that operation should now be sent to
+`https://localhost:5001/Tables` instead. The original endpoint should be stored in an `x-recording-upstream-base-uri` --
+the proxy will send the original request and record the result.
+
+The [`TestProxyManager`][test_proxy_manager] does this for you.
+
+### How does the test proxy know when and what to record or playback?
```

Comment:
For the purpose of migration, do we need to include this section? We can say the TestProxyTestBase handles all of this for client libraries internally and the details can be found in the tools repo documentation.